### PR TITLE
KTIJ-1086 `Replace 'isEmpty' with 'ifEmpty'` quickfix breaks code in case it is performed in a loop with `continue` or `break`

### PR DESCRIPTION
**Pull request by t-kameyama**
Issue link: <https://youtrack.jetbrains.com/issue/KTIJ-1086>

### CODE REVIEW

File: idea/src/org/jetbrains/kotlin/idea/inspections/ReplaceIsEmptyWithIfEmptyInspection.kt [hunk 1]
```diff
@@ -75,6 +77,15 @@ class ReplaceIsEmptyWithIfEmptyInspection : AbstractKotlinInspection() {
             if (selfValueExpression !is KtThisExpression) return
         }
 
+        val loop = ifExpression.getStrictParentOfType<KtLoopExpression>()
+        if (loop != null) {
+            val defaultValueExpression = (if (replacement.negativeCondition) elseExpression else thenExpression)
+            if (defaultValueExpression.anyDescendantOfType<KtExpression> {
+                    (it is KtContinueExpression || it is KtBreakExpression) && it.getStrictParentOfType<KtLoopExpression>() == loop
```
Comment (yanex) [inline idea/src/org/jetbrains/kotlin/idea/inspections/ReplaceIsEmptyWithIfEmptyInspection.kt:84]: Failing case (labeled `break`/`continue`):

```
fun check(listOne: List<String>, listTwo: List<String>) {
    outer@ for (i in 1..10) {
        val z = if (listOne.isNotEmpty()) { //perform `Replace with 'ifEmpty {...}'` quickfix
            listOne
        } else {
            while (3 > 2) {
                listOf(listTwo.firstOrNull() ?: continue@outer)
            }
        }
    }
}
```

**Comment by yanex** [comment]
The bug is still here:

```
// PROBLEM: none
// WITH_RUNTIME
fun test(listOne: List<String>, listTwo: List<String>) {
    outer@ for (i in 1..10) {
        val z = if (listOne.isNotEmpty()) {
            listOne
        } else {
            for (j in 1..10) {
                break@outer
            }
        }
    }
}
```

**Comment by t-kameyama** [comment]
> The bug is still here:

Fixed in 569d3da, PTAL.

**Comment by t-kameyama** [comment]
@yanex 
569d3da has not been merged into the master.

**Comment by yanex** [comment]
Yep. Now everything is merged as should (https://github.com/JetBrains/intellij-kotlin/commit/9432297d26bc004b1506e9786f608f2b0883f961, https://github.com/JetBrains/intellij-kotlin/commit/8d4df6e3a346617d5735571ffdcedbeafdbf68bf, https://github.com/JetBrains/intellij-kotlin/commit/3f9b7235417001123115b2548bf4e3da14e6e860). Thanks for noticing!
And thanks for the contribution!

**Pull request closed without merging**